### DISCUSSION TITLE
Bonsai random hang issue

### DISCUSSION THREAD

**Discussion by AOONIH** [bonsai-rx/bonsai#1228] [Q&A]
There's been problem with Bonsai crashing randomly when used as a state machine. These crashes produce no error message, some workflows still running, but the program is unable to be stopped. I've possibly isolated the bug to subject subscription in SelectMany. Taking the subject out of the SelectMany and piping it in using a property mapping appears to resolve the issue

![Crashing workflow](https://user-images.githubusercontent.com/40366898/212351964-38ea1ff3-95be-44d7-b42c-44efc545ed8d.png)

![Fixed workflow](https://user-images.githubusercontent.com/40366898/212352521-04ef14d3-6733-41c0-a3cc-0612196beec3.png)

**Comment by bruno-f-cruz** [comment]
Can you include the workflow so we can take a look?
Thanks!

**Reply by AOONIH** [reply]
Here is the bonsai workflow
[ProbReward.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/10425682/ProbReward.bonsai.zip)
